lms/djangoapps/courseware/access.py

```diff
@@ -160,11 +181,10 @@ def _can_access_descriptor_with_start_date(user, descriptor, course_key):  # pyl
             descriptor,
             course_key=course_key
         )
-        return (
-            descriptor.start is None
-            or now > effective_start
-            or in_preview_mode()
-        )
+        if descriptor.start is None or now > effective_start or in_preview_mode():
+            return ACCESS_GRANTED
+
+        return StartDateError("{:%B %d, %Y}".format(descriptor.start))
```

Comment:
The formatting of the start date should be handled within the StartDateError object.  We want the StartDateError class to be an abstraction for that error and for all formatting related to that error.  Also, a developer who runs into this error would probably want to look at the original value of the start date.